pkg/dataplane/util/image_registry.go

```diff
@@ -0,0 +1,127 @@
+package util
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"strings"
+
+	mc "github.com/openshift/api/machineconfiguration/v1"
+	ocpimage "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// machineConfigIgnition - holds the relevant parts of the ignition file that we need to create
+// the registries.conf on the dataplane nodes.
+type machineConfigIgnition struct {
+	Ignition struct {
```

Comment:
I guess something similar is defined in MachineConfig api module. Are we intentionally not trying to import the original struct?

Reply:
I thought the same, but it's just defined like this:
https://github.com/openshift/api/blob/master/machineconfiguration/v1/types.go#L320-L322

So I thought just implementing a minimal struct with our requirements would be a cleaner option.